test-utils/runtime/src/extrinsic.rs

```diff
@@ -0,0 +1,197 @@
+		let signature = match self.signature {
+			Some(signature) => signature,
+			None => sp_keyring::AccountKeyring::from_public(&self.transfer.from)
+				.expect("Creates keyring from public key.")
+				.sign(&self.transfer.encode()),
+		};
+		PalletCall::transfer {
+			transfer: self.transfer,
+			signature,
+			exhaust_resources_when_not_first: self.exhaust_resources,
+		}
+	}
+}
+
+/// Generates `Extrinsic`
+pub struct ExtrinsicBuilder {
+	function: RuntimeCall,
+	is_unsigned: bool,
+}
+
+impl ExtrinsicBuilder {
+	/// Create builder for given `RuntimeCall`
+	pub fn new(function: impl Into<RuntimeCall>) -> Self {
+		Self { function: function.into(), is_unsigned: false }
+	}
+
+	/// Create builder for given `Transfer`
+	pub fn new_transfer(transfer: Transfer) -> Self {
+		Self::new(TransferCallBuilder::new(transfer).build())
+	}
+
+	/// Create builder for `PalletCall::authorities_change` call using given parameters
+	pub fn new_authorities_change(new_authorities: Vec<AuthorityId>) -> Self {
+		Self::new(PalletCall::authorities_change { new_authorities })
+	}
+
+	/// Create builder for `PalletCall::include_data` call using given parameters
+	pub fn new_include_data(data: Vec<u8>) -> Self {
+		Self::new(PalletCall::include_data { data })
+	}
+
+	/// Create builder for `PalletCall::storage_change` call using given parameters
+	pub fn new_storage_change(key: Vec<u8>, value: Option<Vec<u8>>) -> Self {
+		Self::new(PalletCall::storage_change { key, value })
+	}
+
+	/// Create builder for `PalletCall::storage_change_unsigned` call using given parameters. Will
+	/// create unsigned Extrinsic.
+	pub fn new_storage_change_unsigned(key: Vec<u8>, value: Option<Vec<u8>>) -> Self {
+		Self::new(PalletCall::storage_change_unsigned { key, value }).unsigned()
+	}
+
+	/// Create builder for `PalletCall::offchain_index_set` call using given parameters
+	pub fn new_offchain_index_set(key: Vec<u8>, value: Vec<u8>) -> Self {
+		Self::new(PalletCall::offchain_index_set { key, value })
+	}
+
+	/// Create builder for `PalletCall::offchain_index_clear` call using given parameters
+	pub fn new_offchain_index_clear(key: Vec<u8>) -> Self {
+		Self::new(PalletCall::offchain_index_clear { key })
+	}
+
+	/// Create builder for `PalletCall::new_store` call using given parameters
+	pub fn new_store(data: Vec<u8>) -> Self {
+		Self::new(PalletCall::store { data })
+	}
+
+	/// Create builder for `PalletCall::new_deposit_log_digest_item` call using given `log`
+	pub fn new_deposit_log_digest_item(log: sp_runtime::generic::DigestItem) -> Self {
+		Self::new(PalletCall::deposit_log_digest_item { log })
+	}
+
+	/// Unsigned `Extrinsic` will be created
+	pub fn unsigned(mut self) -> Self {
+		self.is_unsigned = true;
+		self
+	}
+
+	/// Build `Extrinsic` using embedded parameters
+	pub fn build(self) -> Extrinsic {
+		if self.is_unsigned {
+			Extrinsic::new_unsigned(self.function)
+		} else {
+			let sender = sp_keyring::AccountKeyring::Alice;
```

Comment:
Weird that the builder doesn't expose a function to change this :P But yeah, not a hard requirement and it is for testing any way, so anybody can just add it when required.